src/lib/pubkey/pem/pem.cpp

```diff
@@ -123,6 +123,23 @@ secure_vector<uint8_t> decode(DataSource& source, std::string& label)
    return base64_decode(b64.data(), b64.size());
    }
 
+std::vector<secure_vector<uint8_t>> decode_all(DataSource& source)
+   {
+   std::vector<secure_vector<uint8_t>> pems;
+
+   while(!source.end_of_data())
+      {
+      std::string label;
+      try
+         {
+         pems.push_back(decode(source, label));
+         }
+      catch(const Decoding_Error&) {}
```

Comment:
If the file is completely parsed, but still includes newlines, it will throw when we try to `decode`. Is there a better way to mitigate that?

Reply:
I think PEM interface is due for a design refresh (this code is very old) so that we can indicate errors in a cleaner way. But this works for now.

Anyway we probably want to catch exceptions here because (AIUI) the CA bundle file might include stray text/comments that we'll want to ignore.